File: Copilot/3.Desarrollo Web Endpoint RESTful/python/endpoint.py
class Singleton:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            print("Instancia creada")
            cls._instance = super(Singleton, cls).__new__(cls)
        return cls._instance

# Método de prueba
if __name__ == "__main__":
    a = Singleton()
    b = Singleton()
    print("¿Es la misma instancia? " + str(a is b))

File: ChatGPT/3.Desarrollo Web Endpoint RESTful/Typescript/endpoint.ts
class Singleton {
  private static instance: Singleton;

  private constructor() {
    console.log("Instancia creada");
  }

  public static getInstance(): Singleton {
    if (!Singleton.instance) {
      Singleton.instance = new Singleton();
    }
    return Singleton.instance;
  }

  public sayHello(): void {
    console.log("Hola desde Singleton");
  }
}

// Prueba del Singleton
const instancia1 = Singleton.getInstance();
const instancia2 = Singleton.getInstance();

console.log("¿Ambas instancias son iguales?", instancia1 === instancia2); // true
instancia1.sayHello();
